ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java

```diff
@@ -286,4 +289,47 @@ public void updateInteraction(String interactionId, Map<String, Object> updateCo
             actionListener.onFailure(e);
         }
     }
+
+    /**
+     * Delete interaction with its trace data
+     * @param interactionId interaction id
+     * @param listener callback for delete result
+     */
+    public void deleteInteraction(String interactionId, ActionListener<Boolean> listener) {
+        BulkRequest bulkRequest = new BulkRequest(indexName);
+        bulkRequest.add(new DeleteRequest(indexName, interactionId));
+
+        innerGetTraces(interactionId, ActionListener.wrap(traces -> {
+            traces.forEach(trace -> bulkRequest.add(new DeleteRequest(indexName, trace.getId())));
```

Comment:
Instead of bulk request, can we use delete_by_query to reduce number of operations?

Reply:
make sense, updated to `delete_by_query`